src/overlays.ts

```diff
@@ -0,0 +1,303 @@
+import { AbstractPoint, DistanceOverlayInfo, Distances, Offset } from ".."
+import { ULabelAnnotation } from "./annotation"
+import { get_annotation_class_id } from "./annotation_operators"
+import { ULabelSpatialPayload2D } from "./geometric_utils"
+
+/**
+ * Basic class to hold generic methods useful for creating overlays.
+ */
+class ULabelOverlay {
+    canvas: HTMLCanvasElement
+    context: CanvasRenderingContext2D
+
+    constructor(canvas_width: number, canvas_height: number) {
+        this.createCanvas(canvas_width, canvas_height)
+
+        this.context = this.canvas.getContext("2d")
+
+        this.add_styles()
+    }
+
+    private add_styles() {
+        const css = `
+        /* Very Important. Annotation interaction breaks without this property when overlays are present */
+        .ulabel-overlay {
+            pointer-events: none; 
+        }`
+
+        // Create an id so this specific style tag can be referenced
+        const style_id = "overlay-styles"
+
+        // Don't add the style tag if its already been added once
+        if (document.getElementById(style_id)) return
+
+        // Grab the document's head and create a style tag
+        const head = document.head || document.querySelector("head")
+        const style = document.createElement('style');
+
+        // Add the css and id to the style tag
+        style.appendChild(document.createTextNode(css));
+        style.id = style_id
+
+        // Add the style tag to the document's head
+        head.appendChild(style);
+    }
+
+    public createCanvas(canvas_width, canvas_height): void {
+        // Create the canvas element
+        this.canvas = document.createElement("canvas")
+
+        // Add a class to identify created overlays
+        this.canvas.setAttribute("class", "ulabel-overlay")
+
+        // Overlays should on top of everything, so give it a reasonably large z-index
```

Comment:
What if we want overlays to be over the image but under the annotations? 
Is this easily set via the z-index, and if so, is it easily configurable?

Reply:
Yeah its as simple as changing the z-index. We could add an option when creating an overlay to determine whether it should go in front of or behind annotations.

Reply:
Let's do that. If not in this PR, open an issue to have it configurable. 
Definitely would be necessary if using multiple overlays as well.